### PR TITLE
Explicitly support Rails 8.0

### PR DESCRIPTION
**Pull request by zjohl**
There was some confusion around whether or not this adapter supports Rails 8.0. After testing locally and in CI, it looks like it does so I'm updating the Gemfile accordingly. We'll also run tests on Rails 8.0 going forward

### CODE REVIEW

**Comment by kevin-j-m** [comment]
We should also undo the change in #16 when we introduce this.

File: gemfiles/ar_8_0.gemfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+source "https://rubygems.org"
+gem 'activerecord', '~> 8.0.0'
+gem 'pg', '~> 1.1'
+
+gemspec path: "../"
```
Comment (kevin-j-m) [inline gemfiles/ar_8_0.gemfile:5]: The `Appraisals` file should also be updated to include 7.2 and 8.0

Comment (zjohl) [reply gemfiles/ar_8_0.gemfile:5]: Oooh that's fancy. I see now how I could have solved my past Rails versioning issues

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# Pending
+
+* Explicitly support Rails 8.0
```
Comment (zjohl) [inline CHANGELOG.md:3]: I don't think this is worth an actual release but trying to be proactive about updating this

**Pull request merged**